src/test/java/org/killbill/billing/plugin/adyen/api/TestAdyenPaymentPluginApiHPP.java

```diff
@@ -238,15 +242,53 @@ public void testCancelExpiredPayPalPaymentNoNotification() throws Exception {
+    }
+
+    @Test(groups = "integration",
+          dataProvider = "authResultToTriggerCancelHppPayment")
+    public void testCancelHppPaymentViaCompletionRequest(final String authResult) throws Exception {
+        Payment payment = triggerBuildFormDescriptor(ImmutableMap.<String, String>of(AdyenPaymentPluginApi.PROPERTY_CREATE_PENDING_PAYMENT, "true",
+                                                                                     AdyenPaymentPluginApi.PROPERTY_AUTH_MODE, "true"),
+                                                     TransactionType.AUTHORIZE);
+        PaymentTransactionInfoPlugin trxInfo = adyenPaymentPluginApi.authorizePayment(payment.getAccountId(),
+                                                                                      payment.getId(),
+                                                                                      payment.getTransactions().get(0).getId(),
+                                                                                      payment.getPaymentMethodId(),
+                                                                                      payment.getAuthAmount(),
+                                                                                      payment.getCurrency(),
+                                                                                      PluginProperties.buildPluginProperties(ImmutableMap.<String, String>of("authResult", authResult)),
+                                                                                      context);
+        Assert.assertEquals(trxInfo.getStatus(), PaymentPluginStatus.ERROR);
+        Assert.assertEquals(payment.getTransactions().get(0).getTransactionStatus(), TransactionStatus.PAYMENT_FAILURE);
+        Assert.assertEquals(payment.getTransactions().get(0).getPaymentInfoPlugin().getStatus(), PaymentPluginStatus.ERROR);
+    }
+
     @Test(groups = "integration")
     public void testCancelExpiredBoletoPayment() throws Exception {
         final Payment payment = triggerBuildFormDescriptor(ImmutableMap.<String, String>of(AdyenPaymentPluginApi.PROPERTY_CREATE_PENDING_PAYMENT, "true",
                                                                                            AdyenPaymentPluginApi.PROPERTY_AUTH_MODE, "true"),
                                                            TransactionType.AUTHORIZE);
         dao.updateResponse(payment.getTransactions().get(0).getId(), PaymentServiceProviderResult.PENDING, ImmutableList.<PluginProperty>of(new PluginProperty("paymentMethod", "boletobancario_santander", false)), context.getTenantId());
 
-        final Period expirationPeriod = adyenConfigProperties.getPendingPaymentExpirationPeriod("boletobancario_santander");
-        assertEquals(expirationPeriod.toString(), "P7D");
+        final Period expirationPeriod = adyenConfigProperties.getPendingHppPaymentWithoutCompletionExpirationPeriod();
```

Comment:
Can you explain the change in expectations here?

It doesn't make sense to me: for Boleto bancário, the customer needs a few days to go to a local kiosk in person, and pay the paper voucher.

Reply:
`getPendingHppPaymentWithoutCompletionExpirationPeriod` is used for those HPP payment methods where **no complete auth call** has been received.

To also leverage `paymentMethodToExpirationPeriod`, I expanded the method `getPendingHppPaymentWithoutCompletionExpirationPeriod` to check  `paymentMethodToExpirationPeriod` first. 

`PT50M` is just a hypothetical setting from the test properties file.

Reply:
> `getPendingHppPaymentWithoutCompletionExpirationPeriod` is used for those HPP payment methods where **no complete auth call** has been received.

Yes, and this is what the test is verifying. The voucher has been created (`PENDING` transaction) and no completion call has been received yet (purchaser has yet to go to the store).

> To also leverage `paymentMethodToExpirationPeriod`, I expanded the method `getPendingHppPaymentWithoutCompletionExpirationPeriod` to check `paymentMethodToExpirationPeriod` first.
> 
> `PT50M` is just a hypothetical setting from the test properties file.

Understood, but the tests config being the same, how come the expiration period has changed? The expectations in `TestAdyenConfigProperties` for instance remain the same.

Sorry for being picky here but I want to make sure there is no subtle change in behavior around this (and if the test was badly setup in the first place, I'd rather fix it than expecting _bad_ / _unrealistic_ values).

Reply:
The expectation changes because the expiration of HPP payment method without a complete auth is set via the property of `pendingHppPaymentWithoutCompletionExpirationPeriod`. The change introduced the in the PR #93, but the test was not changed accordingly. 

The reason expectations in TestAdyenConfigProperties didn't change is because the properties are set manually instead of reading from the property file. Without a global `pendingHppPaymentWithoutCompletionExpirationPeriod` being explicitly set, it defaults back to default expiration date by payment method.

Reply:
Are the tests passing for you?

I just pulled `daliwei/0.18.x-backport` as of a361e651e3ce33fa3b42d99175696e979d42b6af and I'm getting:

```
java.lang.AssertionError: expected [PT50M] but found [P7D]
Expected :PT50M
Actual   :P7D
```

Reply:
sorry, pinged the `adyen.properties` file to you offline.

Reply:
Your test configuration has custom properties, hence the expectation changes:

```
org.killbill.billing.plugin.adyen.pendingPaymentExpirationPeriod=P7D
org.killbill.billing.plugin.adyen.pending3DsPaymentExpirationPeriod=PT1H
org.killbill.billing.plugin.adyen.pendingHppPaymentWithoutCompletionExpirationPeriod=PT50M
org.killbill.billing.plugin.adyen.sensitiveProperties=email|shippingAddress|purchaser_email|ip
```

Can you make sure the suites (`-Pmysql` and `-Pintegration-mysql`) pass with just the urls and credentials (rest should use the default config)?

```
org.killbill.billing.plugin.adyen.merchantAccount
org.killbill.billing.plugin.adyen.username
org.killbill.billing.plugin.adyen.password
org.killbill.billing.plugin.adyen.paymentUrl
org.killbill.billing.plugin.adyen.recurringUrl
org.killbill.billing.plugin.adyen.directoryUrl
org.killbill.billing.plugin.adyen.hmac.algorithm
org.killbill.billing.plugin.adyen.hmac.secret
org.killbill.billing.plugin.adyen.skin
org.killbill.billing.plugin.adyen.hpp.target
```

Reply:
@pierre Done in a34e15e

Reply:
Thanks! 🎉